src/AttrValidator.hack

```diff
@@ -137,7 +137,7 @@ class HTMLPurifier_AttrValidator {
                 // delegate it to the attribute classes to say exactly what.
 
                 // simple substitution
-                $attr[$attr_key] = $result;
+                $attr[$attr_key] = (string)$result;
```

Comment:
This if block checks `is_string`: do we need to cast it?

Reply:
I reverted this change back because at line 119, `$result=false`